app/src/test/java/com/example/parkingsystemkotlin/mvp/presenter/ReservationsRecyclerPresenterTest.kt

```diff
@@ -0,0 +1,75 @@
+package com.example.parkingsystemkotlin.mvp.presenter
+
+import com.example.parkingsystemkotlin.mvp.contract.ReservationsRecyclerContract
+import com.example.parkingsystemkotlin.mvp.model.ReservationsRecyclerModel
+import com.example.parkingsystemkotlin.mvp.model.reservation.DataBase
+import com.example.parkingsystemkotlin.mvp.model.reservation.Reservation
+import com.nhaarman.mockito_kotlin.mock
+import com.nhaarman.mockito_kotlin.verify
+import com.nhaarman.mockito_kotlin.whenever
+import java.util.Calendar
+import org.junit.Assert.assertEquals
+import org.junit.Test
+
+
+class ReservationsRecyclerPresenterTest {
+
+    private val reservationInformationDB: DataBase = mock()
+    private val view: ReservationsRecyclerContract.ReservationsRecyclerViewContract = mock()
+    private val model: ReservationsRecyclerContract.ReservationsRecyclerModelContract = ReservationsRecyclerModel(reservationInformationDB)
+    private val presenter: ReservationsRecyclerContract.ReservationsRecyclerPresenterContract = ReservationsRecyclerPresenter(model, view)
+    private val reservation: Reservation = Reservation(
+        getCalendarDate(YEAR_START, MONTH_START, DAY_OF_MONTH_START, HOUR_OF_DAY_START, MINUTE_START),
+        getCalendarDate(YEAR_FINISH, MONTH_FINISH, DAY_OF_MONTH_FINISH, HOUR_OF_DAY_FINISH, MINUTE_FINISH),
+        SECURITY_CODE,
+        PLACE
+    )
+    private val reservationsToPlace: MutableList<Reservation> = mutableListOf()
+
+    @Test
+    fun `see all reservations test`() {
+        //Setup
+        reservationsToPlace.add(reservation)
+        whenever(reservationInformationDB.getAllReservations()).thenReturn(reservationsToPlace)
+        //Execute
+        presenter.seeAllReservations()
+        //Assert
+        verify(view).seeAllReservationsInScreen(model.getAllReservations())
+        assertEquals(reservationsToPlace, model.getAllReservations())
+    }
+
+    @Test
+    fun `on back button clicked test`() {
+        //Execute
+        presenter.onBackButtonClicked()
+        //Assert
+        verify(view).finishActivityWhitBack()
+    }
+
+    private fun getCalendarDate(year: Int, month: Int, dayOfMonth: Int, hourOfDay: Int, minute: Int): Calendar {
+        val calendar: Calendar = Calendar.getInstance()
+        calendar.set(Calendar.DAY_OF_MONTH, dayOfMonth)
+        calendar.set(Calendar.MONTH, month)
+        calendar.set(Calendar.YEAR, year)
+        calendar.set(Calendar.HOUR_OF_DAY, hourOfDay)
+        calendar.set(Calendar.MINUTE, minute)
+        return calendar
```

Comment:
here you can use apply
https://blog.mindorks.com/learn-kotlin-apply-vs-with

Reply:
done

Reply:
done